scripts/zones/Aydeewa_Subterrane/mobs/Pandemonium_Warden.lua

```diff
@@ -314,4 +191,29 @@ end;
 function onMobDeath(mob, player, isKiller)
     -- TODO: Death speech.
     player:addTitle(PANDEMONIUM_QUELLER);
+    
+    -- Despawn pets
+    for i = 0, 1 do
+        for j = 1, 8 do
+            local pet = GetMobByID(petIDs[i][j]);
+            if pet:isSpawned() then
+                pet:despawn();
+            end
+        end
+    end
 end;
+
+-----------------------------------
+-- handlePet
+-----------------------------------
+
+function handlePet(mob, newPet, oldPet, target, modelId)
+    if oldPet:isSpawned() then
+        oldPet:despawn();
+    end
+    newPet:setModelId(modelId);
+    newPet:spawn();
+    newPet:updateEnmity(target);
+    newPet:clone(oldPet);
```

Comment:
Alright, I can just get rid of clone and bake it into handlePet then. I made it a method in case it was ever useful to anyone else but it probably wouldn't be.